src/puppetlabs/puppetdb/queue.clj

```diff
@@ -83,6 +82,10 @@
              :payload (stream->json command-stream)
              :entry entry))))
 
+(defn note-attempt [cmdref exception]
```

Comment:
Hmm, I avoided conj or assoc because I didn't feel like the concrete implementation was germane, but maybe that's not important?